hazelcast/src/test/java/com/hazelcast/partition/strategy/AttributePartitioningStrategyIntegrationTest.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (c) 2008-2023, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.partition.strategy;
+
+import com.hazelcast.config.Config;
+import com.hazelcast.config.IndexConfig;
+import com.hazelcast.config.PartitioningStrategyConfig;
+import com.hazelcast.core.HazelcastException;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.HazelcastJsonValue;
+import com.hazelcast.instance.impl.HazelcastInstanceProxy;
+import com.hazelcast.jet.SimpleTestInClusterSupport;
+import com.hazelcast.map.IMap;
+import com.hazelcast.nio.serialization.ClassDefinition;
+import com.hazelcast.nio.serialization.ClassDefinitionBuilder;
+import com.hazelcast.nio.serialization.genericrecord.GenericRecord;
+import com.hazelcast.nio.serialization.genericrecord.GenericRecordBuilder;
+import com.hazelcast.partition.Partition;
+import com.hazelcast.partition.PartitioningStrategy;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.hazelcast.spi.properties.ClusterProperty.PARTITION_COUNT;
+import static org.junit.Assert.assertEquals;
+
+@RunWith(HazelcastSerialClassRunner.class)
+public class AttributePartitioningStrategyIntegrationTest extends SimpleTestInClusterSupport {
+    private static final ClassDefinition PORTABLE_CLASS = new ClassDefinitionBuilder(1, 1, 1)
+            .addLongField("id")
+            .addStringField("name")
+            .addStringField("org")
+            .build();
+
+    private static final PartitioningStrategy STRATEGY = new AttributePartitioningStrategy("org");
+    private static final String PARTITION_ATTRIBUTE = "1";
+    private static final Object[] PARTITION_KEY = new Object[]{PARTITION_ATTRIBUTE};
+
+    @BeforeClass
+    public static void beforeClass() {
+        Config config = smallInstanceConfig().setProperty(PARTITION_COUNT.getName(), "3");
+        config.getMapConfig("test").setPartitioningStrategyConfig(new PartitioningStrategyConfig()
+                .setPartitioningStrategy(STRATEGY));
+        config.getMapConfig("test").addIndexConfig(new IndexConfig(IndexConfig.DEFAULT_TYPE, "__key.org"));
```

Comment:
the index seem to be not used. Or do we want to test partitioning of the index as well?